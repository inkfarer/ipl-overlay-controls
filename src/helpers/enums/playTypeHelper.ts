import { PlayType } from '../../types/enums/playType';

export class PlayTypeHelper {
    static toPrettyString(value: PlayType, gameCount?: number): string {
        const count = !gameCount ? '' : ` ${gameCount}`;
        return {
            [PlayType.PLAY_ALL]: `Play all${count}`,
            [PlayType.BEST_OF]: `Best of${count}`
        }[value];
    }

    static fromBattlefySeriesStyle(value: string): PlayType | undefined {
        switch (value?.toLowerCase()) {
            case 'gamespermatch':
                return PlayType.PLAY_ALL;
            case 'bestof':
                return PlayType.BEST_OF;
            default:
                return undefined;
        }
    }

    static fromSmashggSetGamesType(value: number): PlayType | undefined {
        switch (value) {
            case 1:
                return PlayType.BEST_OF;
            case 2:
                return PlayType.PLAY_ALL;
            default:
                return undefined;
        }
    }
}
