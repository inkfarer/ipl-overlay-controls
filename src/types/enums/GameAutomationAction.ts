export enum GameAutomationAction {
    END_GAME = 'END_GAME',
    START_GAME = 'START_GAME',
    NONE = 'NONE'
}
