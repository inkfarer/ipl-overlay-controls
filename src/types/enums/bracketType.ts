export enum BracketType {
    SWISS = 'SWISS',
    DOUBLE_ELIMINATION = 'DOUBLE_ELIMINATION',
    SINGLE_ELIMINATION = 'SINGLE_ELIMINATION',
    ROUND_ROBIN = 'ROUND_ROBIN',
    LADDER = 'LADDER'
}

export class BracketTypeHelper {
    static fromBattlefy(type: string, style?: string): BracketType {
        if (type === 'elimination') {
            if (style == null) {
                throw new Error('Got bracket type "elimination" from Battlefy without "style" parameter '
                    + '(single or double?)');
            }

            return {
                single: BracketType.SINGLE_ELIMINATION,
                double: BracketType.DOUBLE_ELIMINATION
            }[style];
        }

        return {
            ladder: BracketType.LADDER,
            swiss: BracketType.SWISS,
            roundrobin: BracketType.ROUND_ROBIN
        }[type];
    }
}
