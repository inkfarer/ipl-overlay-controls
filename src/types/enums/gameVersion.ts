export enum GameVersion {
    SPLATOON_2 = 'SPLATOON_2',
    SPLATOON_3 = 'SPLATOON_3'
}

export class GameVersionHelper {
    static toPrettyString(value: GameVersion): string {
        return {
            [GameVersion.SPLATOON_2]: 'Splatoon 2',
            [GameVersion.SPLATOON_3]: 'Splatoon 3'
        }[value];
    }
}
