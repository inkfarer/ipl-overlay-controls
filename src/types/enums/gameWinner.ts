export enum GameWinner {
    NO_WINNER = 'none',
    ALPHA = 'alpha',
    BRAVO = 'bravo'
}

export type GameWinnerValues = `${GameWinner}`;
