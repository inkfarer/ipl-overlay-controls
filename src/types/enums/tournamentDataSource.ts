export enum TournamentDataSource {
    BATTLEFY = 'BATTLEFY',
    SMASHGG = 'SMASHGG',
    UPLOAD = 'UPLOAD',
    SENDOU_INK = 'SENDOU_INK',
    UNKNOWN = 'UNKNOWN'
}

export class TournamentDataSourceHelper {
    static toPrettyString(value: TournamentDataSource | `${TournamentDataSource}`): string {
        return {
            [TournamentDataSource.BATTLEFY]: 'Battlefy',
            [TournamentDataSource.SMASHGG]: 'Smash.gg',
            [TournamentDataSource.UPLOAD]: 'Uploaded file',
            [TournamentDataSource.SENDOU_INK]: 'sendou.ink',
            [TournamentDataSource.UNKNOWN]: 'Unknown'
        }[value];
    }
}
