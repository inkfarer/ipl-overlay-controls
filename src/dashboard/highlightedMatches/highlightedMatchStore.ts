import { NodeCGBrowser } from 'nodecg/browser';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import {
    HighlightedMatches,
    TournamentData
} from 'schemas';
import { SelectOptions } from '../types/select';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';

const highlightedMatches = nodecg.Replicant<HighlightedMatches>('highlightedMatches');
const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');

export const highlightedMatchReps = [ highlightedMatches, tournamentData ];

export interface HighlightedMatchStore {
    highlightedMatches: HighlightedMatches;
    tournamentData: TournamentData;
}

export const highlightedMatchStore = createStore<HighlightedMatchStore>({
    state: {
        highlightedMatches: null,
        tournamentData: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async getHighlightedMatches(store, { options }: { options: SelectOptions }): Promise<void> {
            const getAllMatches = options.some(option => option.value === 'all');

            if (getAllMatches) {
                return nodecg.sendMessage('getHighlightedMatches', { getAllMatches });
            } else {
                const values = options.map(option => option.value);

                switch (store.state.tournamentData.meta.source) {
                    case TournamentDataSource.BATTLEFY:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            stages: values
                        });
                    case TournamentDataSource.SMASHGG:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            streamIDs: values.map(value => parseInt(value))
                        });
                    default:
                        throw new Error(`Cannot import data from source '${store.state.tournamentData.meta.source}'`);
                }
            }
        },
        async setNextMatch(store, { teamAId, teamBId }: { teamAId: string, teamBId: string }): Promise<void> {
            return nodecg.sendMessage('setNextRound', { teamAId, teamBId });
        }
    }
});

export const highlightedMatchStoreKey: InjectionKey<Store<HighlightedMatchStore>> = Symbol();

export function useHighlightedMatchStore(): Store<HighlightedMatchStore> {
    return useStore(highlightedMatchStoreKey);
}
