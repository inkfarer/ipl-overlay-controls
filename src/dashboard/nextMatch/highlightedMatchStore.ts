import { NodeCGBrowser } from 'nodecg/browser';
import { HighlightedMatches, TournamentData } from 'schemas';
import { SelectOptions } from '../types/select';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { defineStore } from 'pinia';

const highlightedMatches = nodecg.Replicant<HighlightedMatches>('highlightedMatches');
const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');

export const highlightedMatchReps = [ highlightedMatches, tournamentData ];

export interface HighlightedMatchStore {
    highlightedMatches: HighlightedMatches;
    tournamentData: TournamentData;
}

export const useHighlightedMatchStore = defineStore('highlightedMatches', {
    state: () => ({
        highlightedMatches: null,
        tournamentData: null
    } as HighlightedMatchStore),
    actions: {
        async getHighlightedMatches({ options }: { options: SelectOptions }): Promise<void> {
            const getAllMatches = options.some(option => option.value === 'all');

            if (getAllMatches) {
                return nodecg.sendMessage('getHighlightedMatches', { getAllMatches });
            } else {
                const values = options.map(option => option.value);

                switch (this.tournamentData.meta.source) {
                    case TournamentDataSource.BATTLEFY:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            stages: values
                        });
                    case TournamentDataSource.SMASHGG:
                        return nodecg.sendMessage('getHighlightedMatches', {
                            getAllMatches: false,
                            streamIDs: values.map(value => parseInt(value))
                        });
                    default:
                        throw new Error(`Cannot import data from source '${this.tournamentData.meta.source}'`);
                }
            }
        },
        async setNextMatch(data: SetNextRoundRequest): Promise<void> {
            return nodecg.sendMessage('setNextRound', data);
        }
    }
});
