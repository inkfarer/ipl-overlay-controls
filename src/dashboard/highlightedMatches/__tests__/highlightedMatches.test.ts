import HighlightedMatches from '../highlightedMatches.vue';
import { createStore } from 'vuex';
import { HighlightedMatchStore, highlightedMatchStoreKey } from '../highlightedMatchStore';
import difference from 'lodash/difference';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { config, mount } from '@vue/test-utils';

describe('HighlightedMatches', () => {
    config.global.stubs = {
        HighlightedMatchImporter: true,
        HighlightedMatchViewer: true,
        IplErrorDisplay: true
    };

    function createHighlightedMatchStore() {
        return createStore<HighlightedMatchStore>({
            state: {
                tournamentData: null,
                highlightedMatches: null
            }
        });
    }

    const importableSources = [
        TournamentDataSource.SMASHGG,
        TournamentDataSource.BATTLEFY
    ];

    difference(Object.values(TournamentDataSource), importableSources).forEach(source => {
        it(`displays warning for source ${source}`, () => {
            const store = createHighlightedMatchStore();
            store.state.tournamentData = {
                meta: {
                    source,
                    id: 'tournament123'
                },
                teams: []
            };
            const wrapper = mount(HighlightedMatches, {
                global: {
                    plugins: [[store, highlightedMatchStoreKey]]
                }
            });

            expect(wrapper.findComponent('[data-test="unsupported-source-warning"]').isVisible()).toEqual(true);
            expect(wrapper.findComponent('[data-test="unsupported-source-warning"]').text()).toMatchSnapshot();
            expect(wrapper.findComponent('highlighted-match-importer-stub').exists()).toEqual(false);
            expect(wrapper.findComponent('highlighted-match-viewer-stub').exists()).toEqual(false);
        });
    });

    importableSources.forEach(source => {
        it(`displays importer and viewer for source ${source}`, () => {
            const store = createHighlightedMatchStore();
            store.state.tournamentData = {
                meta: {
                    source,
                    id: 'tournament123'
                },
                teams: [],
                stages: [{ name: 'Cool Stage', id: '123123', type: 'SWISS' }]
            };
            const wrapper = mount(HighlightedMatches, {
                global: {
                    plugins: [[store, highlightedMatchStoreKey]]
                }
            });

            expect(wrapper.findComponent('[data-test="unsupported-source-warning"]').exists()).toEqual(false);
            expect(wrapper.findComponent('[data-test="missing-stages-message"]').exists()).toEqual(false);
            expect(wrapper.findComponent('highlighted-match-importer-stub').isVisible()).toEqual(true);
            expect(wrapper.findComponent('highlighted-match-viewer-stub').isVisible()).toEqual(true);
        });

        it(`displays warning when stages are not present for ${source}`, () => {
            const store = createHighlightedMatchStore();
            store.state.tournamentData = {
                meta: {
                    source,
                    id: 'tournament123'
                },
                teams: [],
                stages: undefined
            };
            const wrapper = mount(HighlightedMatches, {
                global: {
                    plugins: [[store, highlightedMatchStoreKey]]
                }
            });

            expect(wrapper.findComponent('[data-test="unsupported-source-warning"]').exists()).toEqual(false);
            expect(wrapper.findComponent('[data-test="missing-stages-message"]').exists()).toEqual(true);
            expect(wrapper.findComponent('highlighted-match-importer-stub').exists()).toEqual(false);
            expect(wrapper.findComponent('highlighted-match-viewer-stub').exists()).toEqual(false);
        });
    });
});
