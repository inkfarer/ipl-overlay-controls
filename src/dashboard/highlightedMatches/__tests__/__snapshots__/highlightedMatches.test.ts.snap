// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedMatches displays warning for source UNKNOWN 1`] = `"Cannot import data from source 'Unknown'."`;

exports[`HighlightedMatches displays warning for source UPLOAD 1`] = `"Cannot import data from source 'Uploaded file'."`;
